internal/clients/clientimpl/osvmatcher/osvmatcher.go

```diff
@@ -19,6 +21,8 @@ const (
 	maxConcurrentRequests = 1000
 )
 
+var goVersionSuffixRegexp = regexp.MustCompile(`^/?(v\d+)`)
```

Comment:
Please add a few examples about what type of url this is supposed to match, and what it is supposed to reject. (Just copying one of the links you posted as reject would be great!)